<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot
  name="urdf7">
  <link
    name="base_link">
    <inertial>
      <origin
        xyz="0.0668152216688416 -0.200623079180796 0.0642523578899035"
        rpy="0 0 0" />
      <mass
        value="3.61780596947193" />
      <inertia
        ixx="0.0234239245030477"
        ixy="1.64464864106653E-11"
        ixz="2.5749870118457E-08"
        iyy="0.0259458335995285"
        iyz="-1.09112120541581E-06"
        izz="0.0489634906731094" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf7/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf7/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link
    name="wlf">
    <inertial>
      <origin
        xyz="3.84374028176149E-05 -0.000183531350603929 0.0139345794392523"
        rpy="0 0 0" />
      <mass
        value="0.100845124180233" />
      <inertia
        ixx="4.8134198720732E-05"
        ixy="2.10787776934373E-09"
        ixz="-7.39018533366798E-08"
        iyy="4.81245754596535E-05"
        iyz="3.52867414569589E-07"
        izz="6.31918614311935E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf7/meshes/wlf.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf7/meshes/wlf.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="wlf_joint"
    type="continuous">
    <origin
      xyz="-0.098528 -0.082236 0.039792"
      rpy="1.5708 0 1.5708" />
    <parent
      link="base_link" />
    <child
      link="wlf" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="wlb">
    <inertial>
      <origin
        xyz="-3.84374028175594E-05 0.000183531350603922 0.0139345794392523"
        rpy="0 0 0" />
      <mass
        value="0.100845124180233" />
      <inertia
        ixx="4.8134198720732E-05"
        ixy="2.10787776933083E-09"
        ixz="7.39018533366996E-08"
        iyy="4.81245754596535E-05"
        iyz="-3.5286741456959E-07"
        izz="6.31918614311935E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf7/meshes/wlb.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf7/meshes/wlb.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="wlb_joint"
    type="continuous">
    <origin
      xyz="-0.098528 -0.31879 0.039966"
      rpy="-1.5708 0 -1.5708" />
    <parent
      link="base_link" />
    <child
      link="wlb" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="wrf">
    <inertial>
      <origin
        xyz="-0.000184301483844296 3.45565282207153E-05 0.0139345794392523"
        rpy="0 0 0" />
      <mass
        value="0.100845124180232" />
      <inertia
        ixx="4.81244908153978E-05"
        ixy="1.90300558831799E-09"
        ixz="3.54348114866728E-07"
        iyy="4.81342833649877E-05"
        iyz="-6.64402715376398E-08"
        izz="6.31918614311933E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf7/meshes/wrf.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf7/meshes/wrf.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="wrf_joint"
    type="continuous">
    <origin
      xyz="0.23147 -0.081353 0.039202"
      rpy="-1.5708 0 1.5708" />
    <parent
      link="base_link" />
    <child
      link="wrf" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="wrb">
    <inertial>
      <origin
        xyz="0.000184301483844296 -3.45565282207222E-05 0.0139345794392523"
        rpy="0 0 0" />
      <mass
        value="0.100845124180232" />
      <inertia
        ixx="4.8124490815398E-05"
        ixy="1.90300558831802E-09"
        ixz="-3.54348114866745E-07"
        iyy="4.81342833649879E-05"
        iyz="6.64402715376388E-08"
        izz="6.31918614311934E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf7/meshes/wrb.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf7/meshes/wrb.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="wrb_joint"
    type="continuous">
    <origin
      xyz="0.23147 -0.3179 0.039376"
      rpy="1.5708 0 -1.5708" />
    <parent
      link="base_link" />
    <child
      link="wrb" />
    <axis
      xyz="0 0 1" />
  </joint>
  <link
    name="link1">
    <inertial>
      <origin
        xyz="0.0744834102664675 -8.16528076263978E-08 0.034222433190242"
        rpy="0 0 0" />
      <mass
        value="0.116923919932247" />
      <inertia
        ixx="2.68471985890524E-05"
        ixy="-1.87187165060631E-07"
        ixz="-2.57502484520588E-08"
        iyy="9.09493476625712E-05"
        iyz="-1.5225499347338E-11"
        izz="9.13661053358226E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf7/meshes/link1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://urdf7/meshes/link1.STL" />
      </geometry>
    </collision>
  </link>
  <joint
    name="link1_joint"
    type="revolute">
    <origin
      xyz="0.10365 -0.20235 0.1523"
      rpy="1.5708 0 -1.5708" />
    <parent
      link="base_link" />
    <child
      link="link1" />
    <axis
      xyz="0 0 1" />
    <limit
      lower="0.695"
      upper="2.424000"
      effort="1"
      velocity="1" />
  </joint>
<link name="link2">
  <inertial>
    <origin xyz="5.35195461837112E-08 0.135652709133402 0.024085541915995" rpy="0 0 0" />
    <mass value="0.170021565693756" />
    <inertia ixx="0.000294086775787843" ixy="-2.16015023974072E-07" ixz="-1.26336994195142E-11" iyy="0.00019237216893058" iyz="4.88090805218435E-07" izz="0.000133214990684317" />
  </inertial>
  <visual>
    <origin xyz="0 0 0" rpy="0 0 0" />
    <geometry>
      <mesh filename="package://urdf7/meshes/link2.STL" />
    </geometry>
    <material name="">
      <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
    </material>
  </visual>
  <collision>
    <origin xyz="0 0 0" rpy="0 0 0" />
    <geometry>
      <mesh filename="package://urdf7/meshes/link2.STL" />
    </geometry>
  </collision>
</link>

<joint name="link2_joint" type="revolute">
  <origin xyz="0.1309 0 0.00685" rpy="0 0 0" />
  <parent link="link1" />
  <child link="link2" />
  <axis xyz="0 0 1" />
  <limit lower="3.14" upper="6.3" effort="1" velocity="1" />
</joint>

<link name="right_gripper">
  <inertial>
    <origin xyz="0.00999999999999979 0.026106310774761 0.00806753933511638" rpy="0 0 0" />
    <mass value="0.0334179709308611" />
    <inertia ixx="1.60467989320076E-05" ixy="-2.24079472418864E-20" ixz="-1.65647783061652E-21" iyy="5.59323091632944E-06" iyz="-5.6324521607037E-07" izz="2.01628382095131E-05" />
  </inertial>
  <visual>
    <origin xyz="0 0 0" rpy="0 0 0" />
    <geometry>
      <mesh filename="package://urdf7/meshes/right_gripper.STL" />
    </geometry>
    <material name="">
      <color rgba="0.298039215686275 0.298039215686275 0.298039215686275 1" />
    </material>
  </visual>
  <collision>
    <origin xyz="0 0 0" rpy="0 0 0" />
    <geometry>
      <mesh filename="package://urdf7/meshes/right_gripper.STL" />
    </geometry>
  </collision>
</link>

<joint name="right_gripper_joint" type="prismatic">
  <origin xyz="-0.01 0.21135 -0.027709" rpy="0 0 0" />
  <parent link="link2" />
  <child link="right_gripper" />
  <axis xyz="0 0 1" />
  <limit lower="0" upper="0.03" effort="1" velocity="1" />
</joint>

<link name="left_gripper">
  <inertial>
    <origin xyz="0.0100000000000002 0.0261063107747611 0.00806753933511646" rpy="0 0 0" />
    <mass value="0.0334179709308611" />
    <inertia ixx="1.60467989320076E-05" ixy="1.88902519236445E-20" ixz="2.84395359457362E-22" iyy="5.59323091632944E-06" iyz="-5.63245216070373E-07" izz="2.01628382095131E-05" />
  </inertial>
  <visual>
    <origin xyz="0 0 0" rpy="0 0 0" />
    <geometry>
      <mesh filename="package://urdf7/meshes/left_gripper.STL" />
    </geometry>
    <material name="">
      <color rgba="0.298039215686275 0.298039215686275 0.298039215686275 1" />
    </material>
  </visual>
  <collision>
    <origin xyz="0 0 0" rpy="0 0 0" />
    <geometry>
      <mesh filename="package://urdf7/meshes/left_gripper.STL" />
    </geometry>
  </collision>
</link>

<joint name="left_gripper_joint" type="prismatic">
  <origin xyz="0.01 0.21135 0.081491" rpy="-3.1416 0 3.1416" />
  <parent link="link2" />
  <child link="left_gripper" />
  <axis xyz="0 0 1" />
  <limit lower="0" upper="0.03" effort="1" velocity="1" />
</joint>
  <transmission name="link1_trans">
	<type>transmission_interface/SimpleTransmission</type>
	<joint name="link1_joint">
		<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
	</joint>
	<actuator name="link1_motor">
		<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		<mechanicalReduction>1</mechanicalReduction>
	</actuator>
  </transmission>
  
    <transmission name="link2_trans">
	<type>transmission_interface/SimpleTransmission</type>
	<joint name="link2_joint">
		<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
	</joint>
	<actuator name="link2_motor">
		<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		<mechanicalReduction>1</mechanicalReduction>
	</actuator>
  </transmission>

    <transmission name="left_gripper_trans">
	<type>transmission_interface/SimpleTransmission</type>
	<joint name="left_gripper_joint">
		<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
	</joint>
	<actuator name="left_gripper_motor">
		<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		<mechanicalReduction>1</mechanicalReduction>
	</actuator>
  </transmission>
  
      <transmission name="right_gripper_trans">
	<type>transmission_interface/SimpleTransmission</type>
	<joint name="right_gripper_joint">
		<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
	</joint>
	<actuator name="right_gripper_motor">
		<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		<mechanicalReduction>1</mechanicalReduction>
	</actuator>
  </transmission>
  
  <gazebo>
	<plugin name="control" filename="libgazebo_ros_control.so">
		<robotNamespace>urdf7</robotNamespace>
	</plugin>
  </gazebo>
  
  <gazebo reference="link1">
	<selfCollide>true</selfCollide>
  </gazebo>

  <gazebo reference="link2">
	<selfCollide>true</selfCollide>
  </gazebo>

  <gazebo reference="left_gripper">
	<selfCollide>true</selfCollide>
  </gazebo>

  <gazebo>
    <plugin name="differential_drive_controller" filename="libgazebo_ros_diff_drive.so">
      <legacyMode>false</legacyMode>
      <alwaysOn>true</alwaysOn>
      <updateRate>10</updateRate>
      <leftJoint>wlb_joint</leftJoint>
      <rightJoint>wrb_joint</rightJoint>
      <wheelSeparation>0.29</wheelSeparation>
      <wheelDiameter>0.03</wheelDiameter>
      <torque>20</torque>
      <commandTopic>cmd_vel</commandTopic>
      <odometryTopic>odom</odometryTopic>
      <odometryFrame>odom</odometryFrame>
      <robotBaseFrame>base_link</robotBaseFrame>
    </plugin>
  </gazebo>
</robot>
